New content (guides/refs/updates to existing for LLM content gaps)

hollyschinsky opened the pull request:
## Description

**Added**

- New [Add-on Architecture](https://developer-stage.adobe.com/express/add-ons/docs/guides/learn/platform_concepts/architecture/) guide with dual-runtime system explanation, SDK imports and usage, and cross-runtime communication patterns.
- New [Add-on Project Anatomy & CLI Templates](https://developer-stage.adobe.com/express/add-ons/docs/guides/getting_started/addon-project-anatomy/) guide with add-on project structure, file organization, and template selection for efficient Adobe Express add-on development.
- New [Add-on Development Terminology](https://developer-stage.adobe.com/express/add-ons/docs/guides/learn/fundamentals/terminology/) guide with standardized definitions, decision matrices, and troubleshooting for Adobe Express Add-on development terminology.
- New practical usage guide for [Using Add-on UI SDK Constants](https://developer-stage.adobe.com/express/add-ons/docs/guides/learn/fundamentals/ui-sdk-constants/), import patterns (via `addOnUISdk.constants.*`) or named import), use case examples, etc
- New practical usage guide for [Using Document Sandbox Constants](https://developer-stage.adobe.com/express/add-ons/docs/guides/learn/fundamentals/document-sandbox-constants/) to guide on the use of constants in the Document Sandbox environment, with use cases and practical examples.
- New ["SDK Fundamentals"](https://developer-stage.adobe.com/express/add-ons/docs/guides/learn/fundamentals/terminology/) section in navigation under "Learn" to group terminology and constants usage guides for better discoverability.
- All new guides have FAQs and other helpful metadata updates for LLM/SEO.

**Updated**
- Restructure of the [Add-on UI SDK Constants Reference](https://developer-stage.adobe.com/express/add-ons/docs/references/addonsdk/addonsdk-constants/) - removed the old html table in favor of linkable sections for each
- Improved content and metadata for SEO and AI assistant optimization.


## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

hollyschinsky committed:
Reworks all addOnUISdk constants ref and explanation, new how-to

hollyschinsky committed:
Lint fixes

hollyschinsky committed:
Fix link

hollyschinsky committed:
Fix link

hollyschinsky committed:
Adds new terminology page and new constants pages

hollyschinsky committed:
Fixed jsx issue

hollyschinsky committed:
Closing alert

hollyschinsky committed:
Misc fixes

hollyschinsky committed:
Updates for accuracy

hollyschinsky committed:
Continued updates

hollyschinsky committed:
Remove old files

hollyschinsky committed:
Continued updates

hollyschinsky committed:
Continued updates

hollyschinsky committed:
Continued updates

hollyschinsky committed:
Updated changelogs and misc

hollyschinsky committed:
Updated desc

hollyschinsky committed:
Fix links

hollyschinsky committed:
Updates

hollyschinsky committed:
Merge branch 'main' into ui-sdk-constants-rework

hollyschinsky committed:
Mark new and misc

hollyschinsky committed:
Marks new content

hollyschinsky committed:
Mark one more

hollyschinsky committed:
Fix alert format

hollyschinsky committed:
Cleanup, flow, cross-refs

hollyschinsky committed:
Fix yaml

hollyschinsky committed:
Reorder, fix content

hollyschinsky committed:
Misc updates

hollyschinsky committed:
Content updates

hollyschinsky committed:
Fixed markdown issue

hollyschinsky committed:
Fixed content

hollyschinsky committed:
Fixed content

hollyschinsky committed:
Clean-up, x-refs etc

hollyschinsky committed:
Proj anatomy updates, changelog, nav

hollyschinsky committed:
fixed up misc

hollyschinsky committed:
Formatting

hollyschinsky committed:
Cleanup constants

hollyschinsky committed:
Fix

hollyschinsky committed:
Fixed linting and spacing

hollyschinsky committed:
Consistency and formatting

hollyschinsky committed:
Updated intros

hollyschinsky committed:
Cross-refs

hollyschinsky committed:
Consistency, links updates

hollyschinsky committed:
Review updates

undavide commented on src/pages/guides/getting_started/addon-project-anatomy.md:
```diff
@@ -0,0 +1,810 @@
+**SWC templates are characterized by:**
+
+- **Vanilla JavaScript** (no UI framework like React)
+- **Lit** for creating custom web components
```
`[**Lit**](https://lit.dev/docs/v1/lit-html/introduction/) for creating custom web components`

undavide commented on src/pages/guides/learn/how_to/handle_selection.md:
```diff
@@ -59,6 +59,26 @@ faq:
     - question: "What are common selection-based actions?"
       answer: "Typical actions include updating properties panels, enabling/disabling tools, applying formatting to selected text, grouping elements, and showing context-appropriate options."
+canonical: true
+ai_assistant_note: "This is the authoritative guide for working with element selections in Adobe Express add-ons. 
```
We don't have Handle Element Selection in the How-to's home page `src/pages/guides/learn/how_to/index.md` (that would be the How-to > About page in the sidebar), nor the entire Document Structure > Manage Pages, which should be also added.
I also noticed that I failed to remove, in the `index.md` the box that says "Code Playground integration is coming soon!"

undavide commented on src/pages/references/addonsdk/addonsdk-constants.md:
```diff
@@ -1,276 +1,631 @@
-</tr>
-</tbody>
-</table>
+A set of constants used throughout the Add-on UI SDK for type-safe development. See the [Add-on UI SDK Constants Guide](../../guides/learn/fundamentals/ui-sdk-constants.md) for practical examples and usage patterns.
```
This is SO AWESOME now!!!! Way much better than the old (EDS-incompatible 🤭) table!!

undavide reviewed (approved):
This is a MASSIVE improvement!!
I've left a few comments here and there, just nits. It's really awesome, thank you so much for writing it.

undavide committed:
Rename MCP Server
Move one level up in the IA
(pending: name in the sidebar, release date in changelogs)

undavide committed:
Rename MCP item in the sidebar

undavide committed:
Remove "new." from "adobe.express.com" URLs

undavide commented on src/pages/guides/getting_started/addon-project-anatomy.md:
````diff
@@ -0,0 +1,810 @@
+**UI code (`ui/index.js`):**
+
+```javascript
+import addOnUISdk from "https://new.express.adobe.com/static/add-on-sdk/sdk.js";
````
Fixed

undavide committed:
Code review #1

undavide committed:
Merge branch 'main' into new-content-fix-gaps

undavide committed:
Fix links

undavide committed:
Code review #2

undavide committed:
Merge branch 'barranca/mcp-ga' into new-content-fix-gaps

padmkris123 commented on src/pages/guides/getting_started/local_development/mcp_server.md:
```diff
@@ -15,15 +15,15 @@ keywords:
   - IDE 
-title: Adobe Express Add-on MCP Server (Beta)
+title: Adobe Express Developer MCP Server
 description: Learn how to set up and use the Adobe Express Add-on MCP Server for enhanced development workflow in compatible editors like Cursor.
```
Rename the server in the description too.

undavide committed:
Update npm package name (MCP)

undavide committed:
New screenshot

undavide committed:
Hide Resize/Rescale APIs

undavide committed:
Update brand guidelines

undavide committed:
Fix broken links
